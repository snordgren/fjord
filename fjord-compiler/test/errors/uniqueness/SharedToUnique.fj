module errors.uniqueness.SharedToUnique

useUnique : Int -* Int
useUnique x =
  x

# needs to error, x is not unique
passShared : Int -> Int
passShared x =
  useUnique x
