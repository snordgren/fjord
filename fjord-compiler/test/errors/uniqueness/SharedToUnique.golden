errors/uniqueness/SharedToUnique.fj:10:13:
   |
10 |   useUnique x
   |             ^^
expected unique value
